packages/core/src/shared/filesystemUtilities.ts

```diff
@@ -246,3 +247,117 @@ export async function setDefaultDownloadPath(downloadPath: string) {
+            }
+        })
+        // test concurrent reads on a file
+        await withFailCtx('writeFileConcurrencyTest', () => fs.writeFile(tmpFile, 'concurrencyTest'))
+        const result = await Promise.all([
+            withFailCtx('readFileConcurrent1', () => fs.readFileText(tmpFile)),
+            withFailCtx('readFileConcurrent2', () => fs.readFileText(tmpFile)),
+            withFailCtx('readFileConcurrent3', () => fs.readFileText(tmpFile)),
+            withFailCtx('readFileConcurrent4', () => fs.readFileText(tmpFile)),
+            withFailCtx('readFileConcurrent5', () => fs.readFileText(tmpFile)),
+        ])
+        if (result.some((text) => text !== 'concurrencyTest')) {
+            throw new Error(`Unexpected concurrent file reads: ${result}`)
+        }
+        // test deleting a file
+        await withFailCtx('deleteFileInitial', () => fs.delete(tmpFile))
+        await withFailCtx('writeFileAfterDelete', () => fs.writeFile(tmpFile, 'afterDelete'))
+        await withFailCtx('readNewFileAfterDelete', async () => {
+            const text = await fs.readFileText(tmpFile)
+            if (text !== 'afterDelete') {
+                throw new Error(`Unexpected file content after writing to deleted file: "${text}"`)
+            }
+        })
+        await withFailCtx('deleteFileFully', () => fs.delete(tmpFile))
+        await withFailCtx('notExistsFileAfterDelete', async () => {
+            const res = await fs.exists(tmpFile)
+            if (res) {
+                throw new Error(`Expected file to NOT exist: "${tmpFile}"`)
+            }
+        })
+
+        // test rename
+        await withFailCtx('writeFileForRename', () => fs.writeFile(tmpFile, 'TestingRename'))
+        const tmpFileRenamed = tmpFile + '.renamed'
+        await withFailCtx('renameFile', () => fs.rename(tmpFile, tmpFileRenamed))
+        await withFailCtx('existsRenamedFile', async () => {
+            const res = await fs.exists(tmpFileRenamed)
+            if (!res) {
+                throw new Error(`Expected RENAMED file to exist: "${tmpFileRenamed}"`)
+            }
+        })
+        await withFailCtx('writeToRenamedFile', async () => fs.writeFile(tmpFileRenamed, 'hello'))
+        await withFailCtx('readFromRenamedFile', async () => {
+            const res = await fs.readFileText(tmpFileRenamed)
+            if (res !== 'hello') {
+                throw new Error(`Expected RENAMED file to be writable: "${tmpFileRenamed}"`)
+            }
+        })
+        await withFailCtx('renameFileReset', () => fs.rename(tmpFileRenamed, tmpFile))
+        await withFailCtx('renameFileResetExists', async () => {
+            const res = await fs.exists(tmpFile)
+            if (!res) {
+                throw new Error(`Expected reverted renamed file to exist: "${tmpFile}"`)
+            }
+        })
+    } finally {
+        await fs.delete(tmpFolder, { recursive: true, force: true })
```

Comment:
did we ever see any problems with delete? Just wondering if we might create a file and then never be able to delete it 😄

Reply:
updated this function to write to the /tmp folder so if we are unable to delete it will eventually be cleaned up